src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,128 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String SQL_CREATE_NEW_ROW
+            = "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String SQL_FIND_ROW_BY_ID = "SELECT * FROM books WHERE id = ?";
+    private static final String SQL_UPDATE_ROW
+            = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+    private static final String SQL_DELETE_ROW_BY_ID = "DELETE FROM books WHERE id = ?";
+    private static final String SQL_GET_ALL_ROWS = "SELECT * FROM books";
+
+    @Override
+    public Book create(Book book) {
+        try (PreparedStatement preparedStatement = ConnectionUtil
+                .getConnection().prepareStatement(SQL_CREATE_NEW_ROW,

```

Comment:
I'm not sure that the connection will be automatically closed when used like:
```
getConnection().prepareStatement()
```

It is better to create separate variables within `try-with-resources` block

**Please apply to all places**